### PR TITLE
MOSIP-34112 code changes

### PR DESCRIPTION
**Pull request by sowmya695**

### CODE REVIEW

File: registration-processor/core-processor/registration-processor-bio-dedupe-stage/src/main/java/io/mosip/registration/processor/biodedupe/stage/BioDedupeProcessor.java [hunk 1]
```diff
@@ -132,6 +133,9 @@ public class BioDedupeProcessor {
 	@Value("${registration.processor.missing.biometric.verification.enabled:true}")
 	private boolean missingBiometricVerificationEnabled;
 
+	@Value("${registration.processor.biometric.autoreject.enabled:false}")
+	private boolean biometricAutorejectEnable;
```
Comment (vishwa-vyom) [inline registration-processor/core-processor/registration-processor-bio-dedupe-stage/src/main/java/io/mosip/registration/processor/biodedupe/stage/BioDedupeProcessor.java]: I think auto reject is two words and camel case to be corrected.

File: registration-processor/core-processor/registration-processor-bio-dedupe-stage/src/main/java/io/mosip/registration/processor/biodedupe/stage/BioDedupeProcessor.java [hunk 2]
```diff
@@ -435,6 +439,36 @@ private void postAbisIdentification(InternalRegistrationStatusDto registrationSt
 					registrationStatusDto.getRegistrationId(), BioDedupeConstants.ABIS_RESPONSE_NULL);
 
 		} else {
+			if (registrationType.equalsIgnoreCase(RegistrationType.UPDATE.name()) && matchedRegIds.size() == 1
```
Comment (vishwa-vyom) [inline registration-processor/core-processor/registration-processor-bio-dedupe-stage/src/main/java/io/mosip/registration/processor/biodedupe/stage/BioDedupeProcessor.java]: what if the matchedRegIds size is 4 and all belongs to the same UIN ?

File: registration-processor/core-processor/registration-processor-bio-dedupe-stage/src/main/java/io/mosip/registration/processor/biodedupe/stage/BioDedupeProcessor.java [hunk 3]
```diff
@@ -435,6 +439,36 @@ private void postAbisIdentification(InternalRegistrationStatusDto registrationSt
 					registrationStatusDto.getRegistrationId(), BioDedupeConstants.ABIS_RESPONSE_NULL);
 
 		} else {
+			if (registrationType.equalsIgnoreCase(RegistrationType.UPDATE.name()) && matchedRegIds.size() == 1
+					&& biometricAutorejectEnable) {
+				Optional<String> optionalMatchedRegId = matchedRegIds.stream().findFirst();
+				String matchedRegId = optionalMatchedRegId.get();
+				String uin = idRepoService.getUinByRid(matchedRegId, utilities.getGetRegProcessorDemographicIdentity());
```
Comment (vishwa-vyom) [inline registration-processor/core-processor/registration-processor-bio-dedupe-stage/src/main/java/io/mosip/registration/processor/biodedupe/stage/BioDedupeProcessor.java]: If the RID is not latest (lets assume last RID for that user was demographic update), will idrepo still give the correct UIN ?

Comment (vishwa-vyom) [reply registration-processor/core-processor/registration-processor-bio-dedupe-stage/src/main/java/io/mosip/registration/processor/biodedupe/stage/BioDedupeProcessor.java]: On this line one more question, what if this RID was partially processed and UIN is not yet associated. In that case. what will the method return ?

File: registration-processor/core-processor/registration-processor-bio-dedupe-stage/src/main/java/io/mosip/registration/processor/biodedupe/stage/BioDedupeProcessor.java [hunk 4]
```diff
@@ -435,6 +439,36 @@ private void postAbisIdentification(InternalRegistrationStatusDto registrationSt
...
+					registrationStatusDto.setSubStatusCode(StatusUtil.BIO_DEDUPE_SUCCESS.getCode());
+					regProcLogger.info(LoggerFileConstant.SESSIONID.toString(),
+							LoggerFileConstant.REGISTRATIONID.toString(), registrationStatusDto.getRegistrationId(),
+							BioDedupeConstants.ABIS_RESPONSE_MATCHING);
+				} else {
+					registrationStatusDto
+							.setLatestTransactionStatusCode(RegistrationTransactionStatusCode.FAILED.toString());
+					object.setIsValid(Boolean.FALSE);
+					registrationStatusDto.setStatusCode(RegistrationStatusCode.REJECTED.name());
+					registrationStatusDto.setStatusComment(StatusUtil.UPDATE_PACKET_BIOMETRICS_NOT_FOUND.getMessage());
```
Comment (vishwa-vyom) [inline registration-processor/core-processor/registration-processor-bio-dedupe-stage/src/main/java/io/mosip/registration/processor/biodedupe/stage/BioDedupeProcessor.java]: This case is not about matching with someone else biometrics ? But message says biometrics not found ?

File: registration-processor/core-processor/registration-processor-bio-dedupe-stage/src/main/java/io/mosip/registration/processor/biodedupe/stage/BioDedupeProcessor.java (hunk 1 above)
Comment (vishwa-vyom) [reply registration-processor/core-processor/registration-processor-bio-dedupe-stage/src/main/java/io/mosip/registration/processor/biodedupe/stage/BioDedupeProcessor.java]: Lets rename as below.
registration.processor.biometrics-update.single-match.auto-reject.enabled
biometricsUpdateSingleMatchAutoReject

File: registration-processor/core-processor/registration-processor-bio-dedupe-stage/src/main/java/io/mosip/registration/processor/biodedupe/stage/BioDedupeProcessor.java [hunk 5]
```diff
@@ -435,6 +439,36 @@ private void postAbisIdentification(InternalRegistrationStatusDto registrationSt
...
+					registrationStatusDto.setSubStatusCode(StatusUtil.UPDATE_PACKET_BIOMETRICS_NOT_FOUND.getCode());
+					regProcLogger.info(LoggerFileConstant.SESSIONID.toString(),
+							LoggerFileConstant.REGISTRATIONID.toString(), registrationStatusDto.getRegistrationId(),
+							BioDedupeConstants.NO_MATCH_FOUND_FOR_UPDATE + registrationStatusDto.getRegistrationId());
+				}
+				
+			}else {
 			registrationStatusDto.setStatusCode(RegistrationStatusCode.FAILED.name());
 			registrationStatusDto.setStatusComment(StatusUtil.BIO_DEDUPE_POTENTIAL_MATCH.getMessage());
```
Comment (vishwa-vyom) [inline registration-processor/core-processor/registration-processor-bio-dedupe-stage/src/main/java/io/mosip/registration/processor/biodedupe/stage/BioDedupeProcessor.java]: How are we planning to handle this case, where there is more than one match and the country does not have a manual adjudication system ?

**Review by vishwa-vyom — CHANGES_REQUESTED** [review]

File: registration-processor/core-processor/registration-processor-bio-dedupe-stage/src/main/java/io/mosip/registration/processor/biodedupe/stage/BioDedupeProcessor.java (hunk 1 above)
Comment (sowmya695) [reply registration-processor/core-processor/registration-processor-bio-dedupe-stage/src/main/java/io/mosip/registration/processor/biodedupe/stage/BioDedupeProcessor.java]: resolved

File: registration-processor/core-processor/registration-processor-bio-dedupe-stage/src/main/java/io/mosip/registration/processor/biodedupe/stage/BioDedupeProcessor.java (hunk 2 above)
Comment (sowmya695) [reply registration-processor/core-processor/registration-processor-bio-dedupe-stage/src/main/java/io/mosip/registration/processor/biodedupe/stage/BioDedupeProcessor.java]: resolved

File: registration-processor/core-processor/registration-processor-bio-dedupe-stage/src/main/java/io/mosip/registration/processor/biodedupe/stage/BioDedupeProcessor.java (hunk 3 above)
Comment (sowmya695) [reply registration-processor/core-processor/registration-processor-bio-dedupe-stage/src/main/java/io/mosip/registration/processor/biodedupe/stage/BioDedupeProcessor.java]: resolved

File: registration-processor/core-processor/registration-processor-bio-dedupe-stage/src/main/java/io/mosip/registration/processor/biodedupe/stage/BioDedupeProcessor.java (hunk 4 above)
Comment (sowmya695) [reply registration-processor/core-processor/registration-processor-bio-dedupe-stage/src/main/java/io/mosip/registration/processor/biodedupe/stage/BioDedupeProcessor.java]: resolved

File: registration-processor/core-processor/registration-processor-bio-dedupe-stage/src/main/java/io/mosip/registration/processor/biodedupe/stage/BioDedupeProcessor.java (hunk 5 above)
Comment (sowmya695) [reply registration-processor/core-processor/registration-processor-bio-dedupe-stage/src/main/java/io/mosip/registration/processor/biodedupe/stage/BioDedupeProcessor.java]: resolved

**Review by vishwa-vyom — APPROVED** [review]
Approving this PR without resolving the open comments since the PR is to the feature branch. 
Once the code moves from feature branch to the develop, we have ensure all these comment responses to be reviewed and closed.

**Pull request merged**